dali/python/nvidia/dali/external_source.py

```diff
@@ -56,7 +56,7 @@ def _check_data_batch(data, batch_size, layout):
             raise RuntimeError("The layout '{}' cannot describe {}-dimensional data".format(layout, dim))
 
 
-def _prep_data_for_feed_input(data, batch_size, layout, device_id=None):
+def _prep_data_for_feed_input(data, batch_size, layout, device_id=-1):
```

Comment:
AFAIK, `None` and `-1` have different meanings in the API. Are we sure, that this won't break existing CPU-only code?

Reply:
OK, I guess for my usage it can remain `None` and I will just use `-1` where necessary.